edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CombinedKubernetesConfigProvider.cs

```diff
@@ -9,29 +9,55 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes
     using Microsoft.Azure.Devices.Edge.Agent.Core;
     using Microsoft.Azure.Devices.Edge.Agent.Docker;
     using Microsoft.Azure.Devices.Edge.Agent.Docker.Models;
+    using Microsoft.Azure.Devices.Edge.Agent.Kubernetes.EdgeDeployment;
     using Microsoft.Azure.Devices.Edge.Util;
     using Microsoft.Extensions.Configuration;
     using Newtonsoft.Json;
-    using AuthConfig = global::Docker.DotNet.Models.AuthConfig;
 
-    public class CombinedKubernetesConfigProvider : CombinedDockerConfigProvider
+    public class CombinedKubernetesConfigProvider : ICombinedConfigProvider<CombinedDockerConfig>, ICombinedConfigProvider<CombinedKubernetesConfig>
     {
         readonly IConfigSource configSource;
 
-        public CombinedKubernetesConfigProvider(IEnumerable<AuthConfig> authConfigs, IConfigSource configSource)
-            : base(authConfigs)
+        readonly CombinedDockerConfigProvider dockerConfigProvider;
+
+        public CombinedKubernetesConfigProvider(IEnumerable<global::Docker.DotNet.Models.AuthConfig> authConfigs, IConfigSource configSource)
         {
             this.configSource = Preconditions.CheckNotNull(configSource, nameof(configSource));
+            this.dockerConfigProvider = new CombinedDockerConfigProvider(authConfigs);
         }
 
-        static CreateContainerParameters CloneOrCreateParams(CreateContainerParameters createOptions) =>
-            createOptions != null
-            ? JsonConvert.DeserializeObject<CreateContainerParameters>(JsonConvert.SerializeObject(createOptions))
-            : new CreateContainerParameters();
+        CombinedDockerConfig ICombinedConfigProvider<CombinedDockerConfig>.GetCombinedConfig(IModule module, IRuntimeInfo runtimeInfo)
+            => this.GetCombinedConfig(module, runtimeInfo);
 
-        public override CombinedDockerConfig GetCombinedConfig(IModule module, IRuntimeInfo runtimeInfo)
+        CombinedKubernetesConfig ICombinedConfigProvider<CombinedKubernetesConfig>.GetCombinedConfig(IModule module, IRuntimeInfo runtimeInfo)
         {
-            CombinedDockerConfig combinedConfig = base.GetCombinedConfig(module, runtimeInfo);
+            CombinedDockerConfig dockerConfig = this.GetCombinedConfig(module, runtimeInfo);
+
+            CreatePodParameters createOptions = new CreatePodParameters
+            {
+                Env = dockerConfig.CreateOptions.Env,
+                Image = dockerConfig.CreateOptions.Image,
+                Labels = dockerConfig.CreateOptions.Labels,
+                ExposedPorts = dockerConfig.CreateOptions.ExposedPorts,
+                HostConfig = dockerConfig.CreateOptions.HostConfig,
+                NetworkingConfig = dockerConfig.CreateOptions.NetworkingConfig,
+            };
+
+            // if (this.features.EnableKubernetesExtensions)
```

Comment:
Cleanup?

Reply:
This has been removed by later changesets